### ISSUE TITLE
Navigation Block: Pressing Enter Doesn't Bring Focus To Menu Item(s)

### ISSUE THREAD

**Issue by armandovias** [WordPress/gutenberg#70140]
### Description

In previous WordPress versions before WordPress 6.8, I was able to use the escape key and navigate blocks using the arrow keys and pressing enter to select a menu item in the navigation block to edit it.

However, it's doing the same behavior of #70047.

### Step-by-step reproduction instructions

1. Edit the site and go to a template with a navigation menu.
2. Find an editable block, such as an HTML block and press enter.
3. Press Alt+Shift+O to open up the Block Navigation Structure.
4. Go To The Navigation Block by using the arrow keys and try to go inside the block by pressing the right arrow.
5. Try moving around and press enter to select an item.
6. The focus doesn't move to that particular item.

### Environment info

- WordPress 6.8.1
- Mozilla Firefox (Windows 11)
- NVDA (Non-Visual Desktop Access) (Windows 11)
- Google Chrome (Windows 11)

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
Thanks for the report.

I added several types of blocks inside the Navigation block and tested how the NVDA screen reader reads them. In my testing, the screen reader reading is exactly the same in 6.8.1 and 6.7.2, and the blocks are correctly focused.

- Open the List View and use the right arrow to navigate to the Navigation block.
- Press the right arrow to open the Navigation block.
- Press the down arrow to navigate to one of the items.
- The screen reader should read, for example:
  ```
  Sample Page  not selected  row 8
  Sample Page  link  Block 1 of 2, Level 7.
  ```
- Press the Enter key.
- The screen reader should read, for example:
  ```
  Editor content  region
  Editor canvas  frame
  document   
  Block: Header  document
  Block: Group  document
  Block: Group  document
  Block: Row  document
  Block: Row  document
  Block: Navigation  document  Navigation Menu: "Navigation"
  Block: Page Link  document
  Navigation link text  edit  multi line  Sample Page
  ```

Is there any step I'm missing?

**Comment by armandovias** [comment]
Update: I just tested again a few minutes ago, and it seems to work as expected! Thank you!
